assessment-api/assessment-actors/src/main/scala/org/sunbird/managers/CopyManager.scala

```diff
@@ -0,0 +1,409 @@
+    val oldToNewIdMap = generateOldToNewIdMap(nodeBLRecord, identifiers)
+    nodeBLRecord.keySet().asScala.toList.map(id => {
+      val nodeInfo = nodeBLRecord.get(id).asInstanceOf[util.HashMap[String, AnyRef]]
+      val node = nodesModified.get(id).asInstanceOf[util.HashMap[String, AnyRef]]
+      val nodeMetaData = node.get(AssessmentConstants.METADATA).asInstanceOf[util.HashMap[String, AnyRef]]
+      val newId = identifiers.get(id)
+      if (nodeInfo.get(AssessmentConstants.CONTAINS_BL).asInstanceOf[Boolean]) {
+        val branchingLogic = nodeInfo.get(AssessmentConstants.BRANCHING_LOGIC).asInstanceOf[util.HashMap[String, AnyRef]]
+        branchingLogicModifier(branchingLogic, oldToNewIdMap)
+        nodeMetaData.put(AssessmentConstants.BRANCHING_LOGIC, branchingLogic)
+      }
+      node.remove(AssessmentConstants.IS_NEW)
+      node.put(AssessmentConstants.IS_NEW, false.asInstanceOf[AnyRef])
+      nodesModified.remove(id)
+      nodesModified.put(newId, node)
+    })
+    hierarchy.keySet().asScala.toList.map(id => {
+      val nodeHierarchy = hierarchy.get(id).asInstanceOf[util.HashMap[String, AnyRef]]
+      val children = nodeHierarchy.get(AssessmentConstants.CHILDREN).asInstanceOf[util.ArrayList[String]]
+      val newChildrenList = new util.ArrayList[String]
+      children.map(identifier => {
+        if (identifiers.containsKey(identifier)) newChildrenList.add(identifiers.get(identifier)) else newChildrenList.add(identifier)
+      })
+      nodeHierarchy.remove(AssessmentConstants.CHILDREN)
+      nodeHierarchy.put(AssessmentConstants.CHILDREN, newChildrenList)
+      if (identifiers.containsKey(id)) {
+        hierarchy.remove(id)
+        hierarchy.put(identifiers.get(id), nodeHierarchy)
+      }
+    })
+    request
+  }
+
+  def updateHierarchy(request: Request, node: Node, originNode: Node, originHierarchy: util.Map[String, AnyRef], copyType: String)(implicit ec: ExecutionContext, oec: OntologyEngineContext): Future[Node] = {
+    prepareHierarchyRequest(originHierarchy, originNode, node, copyType, request).map(req => {
+      val hierarchyRequest = new Request(request)
+      hierarchyRequest.putAll(req)
+      val nodesModified: java.util.HashMap[String, AnyRef] = hierarchyRequest.getRequest.get(HierarchyConstants.NODES_MODIFIED).asInstanceOf[java.util.HashMap[String, AnyRef]]
+      val nodeBLRecord = generateNodeBLRecord(nodesModified)
+      val originalRequest = JsonUtils.deserialize(ScalaJsonUtils.serialize(hierarchyRequest), classOf[Request])
+      val BLExists = nodeBLRecord.exists(BLRecord => BLRecord._2.asInstanceOf[util.HashMap[String, AnyRef]].get(AssessmentConstants.CONTAINS_BL) == true)
+      val (firstUpdateRequest, secondUpdateRequest) = if (BLExists) (hierarchyRequest, JsonUtils.deserialize(ScalaJsonUtils.serialize(hierarchyRequest), classOf[Request])) else (originalRequest, new Request())
```

Comment:
Please change the name from firstUpdateRequest to updateHierarchyReq & secondUpdateRequest to branchingUpdateReq